### ISSUE TITLE
could not find RelOptInfo for given relids

### ISSUE THREAD

**Issue by guofengrichard** [greenplum-db/gpdb#6124]
### Greenplum version or build
Master

```
create temp table tt4x(c1 int, c2 int, c3 int);

explain (costs off)
select * from tt4x t1
where not exists (
  select 1 from tt4x t2
    left join tt4x t3 on t2.c3 = t3.c1
    left join ( select t5.c1 as c1
                from tt4x t4 left join tt4x t5 on t4.c2 = t5.c1
              ) a1 on t3.c2 = a1.c1
  where t1.c1 = t2.c2
);

ERROR:  could not find RelOptInfo for given relids (selfuncs.c:5284)
```

**Comment by hlinnaka** [comment]
This was fixed in PostgreSQL 9.4.2. I bisected the fix to:
```
commit 5f3d1909c58aee1911d4b1e7f643bb8db09e8da6
Author: Tom Lane <tgl@sss.pgh.pa.us>
Date:   Sat Apr 25 16:44:27 2015 -0400

    Prevent improper reordering of antijoins vs. outer joins.
    
    An outer join appearing within the RHS of an antijoin can't commute with
    the antijoin, but somehow I missed teaching make_outerjoininfo() about
    that.  In Teodor Sigaev's recent trouble report, this manifests as a
    "could not find RelOptInfo for given relids" error within eqjoinsel();
    but I think silently wrong query results are possible too, if the planner
    misorders the joins and doesn't happen to trigger any internal consistency
    checks.  It's broken as far back as we had antijoins, so back-patch to all
    supported branches.

```

**Comment by guofengrichard** [comment]
Great! Thanks. Will close this issue once we merged up to 9.4 stable.

**Comment by guofengrichard** [comment]
Fixed in GPDB master as we have merged up to 9.4.20.
